src/Transport/Administration/SubscriptionManager.cs

```diff
@@ -14,83 +15,88 @@ class SubscriptionManager : ISubscriptionManager
         static readonly ILog Logger = LogManager.GetLogger<SubscriptionManager>();
 
         readonly AzureServiceBusTransport transportSettings;
-        readonly ServiceBusAdministrationClient administrativeClient;
-        readonly NamespacePermissions namespacePermissions;
+        readonly ServiceBusClient client;
         readonly string subscribingQueue;
         readonly string subscriptionName;
 
         public SubscriptionManager(
             string subscribingQueue,
             AzureServiceBusTransport transportSettings,
-            ServiceBusAdministrationClient administrativeClient,
-            NamespacePermissions namespacePermissions)
+            ServiceBusClient client)
         {
             this.subscribingQueue = subscribingQueue;
             this.transportSettings = transportSettings;
-            this.administrativeClient = administrativeClient;
-            this.namespacePermissions = namespacePermissions;
+            this.client = client;
 
             subscriptionName = transportSettings.SubscriptionNamingConvention(subscribingQueue);
         }
 
         public async Task SubscribeAll(MessageMetadata[] eventTypes, ContextBag context, CancellationToken cancellationToken = default)
         {
-            await namespacePermissions.CanManage(cancellationToken).ConfigureAwait(false);
+            if (eventTypes.Length == 0)
+            {
+                return;
+            }
 
-            foreach (var eventType in eventTypes)
+            var ruleManager = client.CreateRuleManager(transportSettings.Topology.TopicToSubscribeOn, subscriptionName);
+            await using (ruleManager.ConfigureAwait(false))
             {
-                await SubscribeEvent(administrativeClient, eventType.MessageType, cancellationToken).ConfigureAwait(false);
+                if (eventTypes.Length == 1)
```

Comment:
So this is to avoid a performance hit when using Task.WhenAll? (Is it worth it given that we only execute this once at startup)

Reply:
Subscribe is called during runtime as well with an array with one entry.